CMakeLists.txt

```diff
@@ -283,3 +287,78 @@ install(FILES "distribution/linux/openrct2-scenario.desktop" DESTINATION "${CMAK
 install(FILES "distribution/linux/openrct2-uri.desktop" DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/applications")
 install(FILES "distribution/linux/openrct2-mimeinfo.xml" DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/mime/packages/" RENAME "openrct2.xml")
 install(DIRECTORY "distribution/man/" DESTINATION "${CMAKE_INSTALL_MANDIR}/man6" FILES_MATCHING PATTERN "*.6")
+
+if(NOT DISABLE_GUI)
+
+    set(BUNDLE_RESOURCE "${PROJECT}.app/Contents/Resources")
+    set(SOURCE_DATA_DIR "${CMAKE_CURRENT_SOURCE_DIR}/data")
+    set(SOURCE_ICON_DIR "${CMAKE_CURRENT_SOURCE_DIR}/resources/logo")
+    set(ICON_TARGET "${CMAKE_CURRENT_SOURCE_DIR}/distribution/macos/AppIcon.iconset")
+    set(ICON_OUTPUT "${CMAKE_CURRENT_SOURCE_DIR}/distribution/macos/AppIcon.icns")
+
+    add_custom_command(OUTPUT ${ICON_OUTPUT}
+        COMMAND sips -z 16 16     icon_x1024.png --out ${ICON_TARGET}/icon_16x16.png
+        COMMAND sips -z 32 32     icon_x1024.png --out ${ICON_TARGET}/icon_16x16@2x.png
+        COMMAND sips -z 32 32     icon_x1024.png --out ${ICON_TARGET}/icon_32x32.png
```

Comment:
Instead of creating and bundling the same image file twice, you can just name it `icon_32x32.png` and refer to that file both from `16x16@2x` and `32x32@1x` in the `AppIcon.iconset/Contents.json` file:

```json
    ...
      "scale" : "1x"
    },
    {
      "size" : "16x16",
      "idiom" : "mac",
      "filename" : "icon_32x32.png",
      "scale" : "2x"
    },
    {
      "size" : "32x32",
      "idiom" : "mac",
      "filename" : "icon_32x32.png",
      "scale" : "1x"
    },
    {
      "size" : "32x32",
    ...
```